src/python/pants/backend/python/macros/poetry_requirements.py

```diff
@@ -476,14 +467,18 @@ def generate_tgt(parsed_req: PipRequirement) -> PythonRequirementTarget:
                 Dependencies.alias: [file_tgt.address.spec],
                 **tgt_overrides,
             },
-            generator.address.create_generated(parsed_req.project_name),
+            request.template_address.create_generated(parsed_req.project_name),
         )
 
     result = tuple(generate_tgt(requirement) for requirement in requirements) + (file_tgt,)
 
+    if len(result) > 1:
+        # Validate that the resolve is legal.
+        result[0][PythonRequirementResolveField].normalized_value(python_setup)
```

Comment:
w/ overrides, do all tgts need validation?

Reply:
...mm. Yes, they do. Thanks.

Reply:
But... is checking eagerly like this actually necessary? In order to consume targets, this value will be checked... and a `python_requirement` target isn't receiving eager validation.